sdk/reservations/Azure.ResourceManager.Reservations/tests/Tests/CalculatePriceTests.cs

```diff
@@ -0,0 +1,125 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Azure.Core.TestFramework;
+using Azure.ResourceManager.Reservations.Models;
+using Azure.ResourceManager.Resources;
+using NUnit.Framework;
+
+namespace Azure.ResourceManager.Reservations.Tests
+{
+    public class CalculatePriceTests : ReservationsManagementClientBase
+    {
+        private TenantResource Tenant { get; set; }
+
+        public CalculatePriceTests(bool isAsync) : base(isAsync)
+        {
+        }
+
+        [SetUp]
+        public async Task ClearAndInitialize()
+        {
+            if (Mode == RecordedTestMode.Record || Mode == RecordedTestMode.Playback)
+            {
+                await InitializeClients();
+
+                AsyncPageable<TenantResource> tenantResourcesResponse = ArmClient.GetTenants().GetAllAsync();
+                List<TenantResource> tenantResources = await tenantResourcesResponse.ToEnumerableAsync();
+                Tenant = tenantResources.ToArray()[0];
+            }
+        }
+
+        [TestCase]
+        [RecordedTest]
+        public async Task TestCalculatePriceForSharedScopeMonthly()
+        {
+            var billingPlan = "Monthly";
+            var response = await Tenant.CalculateReservationOrderAsync(CreatePurchaseRequestContent("Shared", billingPlan));
+            testCalculatePriceResponse(response, billingPlan);
+        }
+
+        [TestCase]
+        [RecordedTest]
+        public async Task TestCalculatePriceForSharedScopeUpfront()
+        {
+            var billingPlan = "Upfront";
+            var response = await Tenant.CalculateReservationOrderAsync(CreatePurchaseRequestContent("Shared", billingPlan));
+            testCalculatePriceResponse(response, billingPlan);
+        }
+
+        [TestCase]
+        [RecordedTest]
+        public async Task TestCalculatePriceForSingleScopeMonthly()
+        {
+            var billingPlan = "Monthly";
+            var response = await Tenant.CalculateReservationOrderAsync(CreatePurchaseRequestContent("Single", billingPlan));
+            testCalculatePriceResponse(response, billingPlan);
+        }
+
+        [TestCase]
+        [RecordedTest]
+        public async Task TestCalculatePriceForSingleScopeUpfront()
+        {
+            var billingPlan = "Upfront";
+            var response = await Tenant.CalculateReservationOrderAsync(CreatePurchaseRequestContent("Single", billingPlan));
+            testCalculatePriceResponse(response, billingPlan);
+        }
+
+        private void testCalculatePriceResponse(Response<CalculatePriceResponse> response, string billingPlan)
```

Comment:
Based on the .NET convention, I think this should use `PascalCase`.
```suggestion
        private void TestCalculatePriceResponse(Response<CalculatePriceResponse> response, string billingPlan)
```